### PR TITLE
Adding data connectors APIs

### PR DESCRIPTION
**Pull request by nirgfani**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
             "get": {
                 "x-ms-examples": {
                     "Get all scheduled alert rules.": {
-                        "$ref": "./examples/GetScheduledAlertRules.json"
+                        "$ref": "./examples/alertRulses/GetScheduledAlertRules.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Rulses -> Rules

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 2]
```diff
@@ -111,7 +111,7 @@
             "get": {
                 "x-ms-examples": {
                     "Get a scheduled alert rule.": {
-                        "$ref": "./examples/GetScheduledAlertRuleById.json"
+                        "$ref": "./examples/alertRulses/GetScheduledAlertRuleById.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Rulses -> Rules

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 3]
```diff
@@ -157,7 +157,7 @@
             "put": {
                 "x-ms-examples": {
                     "Creates or updates a scheduled alert rule.": {
-                        "$ref": "./examples/CreateScheduledAlertRule.json"
+                        "$ref": "./examples/alertRulses/CreateScheduledAlertRule.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Rulses -> Rules

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 4]
```diff
@@ -212,7 +212,7 @@
             "delete": {
                 "x-ms-examples": {
                     "Delete a scheduled alert rule.": {
-                        "$ref": "./examples/DeleteScheduledAlertRule.json"
+                        "$ref": "./examples/alertRulses/DeleteScheduledAlertRule.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Rulses -> Rules

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 5]
```diff
@@ -255,6 +255,215 @@
...
+            "get": {
+                "x-ms-examples": {
+                    "Get an Office365 data connector.": {
+                        "$ref": "./examples/dataConnectors/GetOfficeDataConnetorById.json"
+                    },
+                    "Get a TI data connector.": {
+                        "$ref": "./examples/dataConnectors/GetThreatIneleganceById.json"
+                    },
+                    "Get a MCAS data connector.": {
+                        "$ref": "./examples/dataConnectors/GetMicrosoftCloudAappSecurityById.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Aapp -> App

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 6]
```diff
@@ -506,6 +715,290 @@
...
+            "type": "object",
+            "description": "Describes an Azure resource with kind.",
+            "properties": {
+                "kind": {
+                    "type": "string",
+                    "description": "The kind of the data connector",
+                    "enum": [
+                        "AzureActiveDirectory",
+                        "AzureSecurityCenter",
+                        "MicrosoftCloudAappSecurity",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Aapp -> App

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 7]
```diff
@@ -506,6 +715,290 @@
...
+                        "modelAsString": true,
+                        "values": [
+                            {
+                                "value": "AzureActiveDirectory"
+                            },
+                            {
+                                "value": "AzureSecurityCenter"
+                            },
+                            {
+                                "value": "MicrosoftCloudAappSecurity"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Aapp -> App

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 8]
```diff
@@ -506,6 +715,290 @@
...
+            "type": "object",
+            "description": "Represents ASC (Azure Security Center) data connector.",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/DataConnectorWithAlerts"
+                }
+            ]
+        },
+        "MCASDataConnector": {
+            "x-ms-discriminator-value": "MicrosoftCloudAappSecurity",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Aapp -> App

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 9]
```diff
@@ -506,6 +715,290 @@
...
+            "type": "object",
+            "description": "Represents MCAS (Microsoft Cloud Aapp Security) data connector.",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Aapp -> App

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 10]
```diff
@@ -506,6 +715,290 @@
...
+                }
+            }
+        },
+        "DataConnectorDataTypeCommon": {
+            "type": "object",
+            "description": "Common field for data type in data connectors.",
+            "properties": {
+                "state": {
+                    "type": "string",
+                    "description": "Describe whther this data type connection is enabled or not.",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: whther -> whether

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 11]
```diff
@@ -594,6 +1087,24 @@
                 "$ref": "#/definitions/AlertRule"
             },
             "x-ms-parameter-location": "method"
+        },
+        "DataConnectorId": {
+            "name": "dataConnectorId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Connecto ID",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Connecto -> Connector

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 12]
```diff
@@ -255,6 +255,215 @@
...
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectors/{dataConnectorId}": {
+            "get": {
+                "x-ms-examples": {
+                    "Get an Office365 data connector.": {
+                        "$ref": "./examples/dataConnectors/GetOfficeDataConnetorById.json"
+                    },
+                    "Get a TI data connector.": {
+                        "$ref": "./examples/dataConnectors/GetThreatIneleganceById.json"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Inelegance -> Intelligence?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 13]
```diff
@@ -506,6 +715,290 @@
...
+            "description": "Describes an Azure resource with kind.",
+            "properties": {
+                "kind": {
+                    "type": "string",
+                    "description": "The kind of the data connector",
+                    "enum": [
+                        "AzureActiveDirectory",
+                        "AzureSecurityCenter",
+                        "MicrosoftCloudAappSecurity",
+                        "ThreatInelegance",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Inelegance -> Intelligence?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 14]
```diff
@@ -506,6 +715,290 @@
...
+                                "value": "AzureActiveDirectory"
+                            },
+                            {
+                                "value": "AzureSecurityCenter"
+                            },
+                            {
+                                "value": "MicrosoftCloudAappSecurity"
+                            },
+                            {
+                                "value": "ThreatInelegance"
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Inelegance -> Intelligence?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 15]
```diff
@@ -506,6 +715,290 @@
...
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/DataConnectorDataTypeCommon"
+                        }
+                    ]
+                }
+            }
+        },
+        "TIDataConnector": {
+            "x-ms-discriminator-value": "ThreatInelegance",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Inelegance -> Intelligence?

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 16]
```diff
@@ -506,6 +715,290 @@
...
+            "type": "object",
+            "description": "Represents threat inelegance data connector.",
```
Comment (nschonni) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: Inelegance -> Intelligence?

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Please check the possible typos - otherwise LGTM.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json [hunk 17]
```diff
@@ -764,8 +764,8 @@
                     "enum": [
                         "AzureActiveDirectory",
                         "AzureSecurityCenter",
-                        "MicrosoftCloudAappSecurity",
-                        "ThreatInelegance",
+                        "MicrosoftCloudAppSecurity",
```
Comment (jianghaolu) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: @nirgfani so these are actually typos? The service will actually return these values? Just want to double check before this goes into master.

Comment (nirgfani) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json]: They were typos. It wasn't auto generated.
Thanks for catching them all.

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**